esp-hal/src/gpio/mod.rs

```diff
@@ -987,6 +1007,27 @@ macro_rules! io_type {
     (Output, $gpionum:literal) => {
         impl $crate::gpio::OutputPin for GpioPin<$gpionum> {}
     };
+    (Analog, $gpionum:literal) => {
+        // FIXME: the implementation shouldn't be in the GPIO module
```

Comment:
Where else would you put this?

Reply:
The implementation itself is target-specific, it has no place in gpio/mod.rs. I don't yet know how to organize it, though, but I want to keep this macro just a bunch of "if this, then impl this trait in one line" cases, like Input and Output are right now.